### PR TITLE
Add metrics for server messages sent

### PR DESCRIPTION
**Pull request by prburgu**
### What is the purpose of this pull request?
This PR adds metrics for the total number of messages sent and failed to send to clients. We'll be using these metrics monitor how many messages are being sent to clients and if there are any failures sending. This gives us signal whether service is functioning properly or having issues.

### What changes did you make? (overview)
Registered a new counter metrics and incremented the counters upon sending the message to clients.

### Checklist

- [ ] I've added tests for this change
- [x] I've added a Changelog entry
- [x] I've updated documentation

### CODE REVIEW

File: node/session.go
Comment (sponomarev) [inline node/session.go:143]: Is it a real case when `s.node` is `nil` in this context?

Comment (prburgu) [reply node/session.go:143]: I wasn't expecting s.node to be nil in this context, however [the tests were failing](https://github.com/anycable/anycable-go/runs/2220595329) without this check, so it seems there are cases where s.node could be nil in this context.

Comment (sponomarev) [reply node/session.go:143]: I guess it's only a case for tests, but I would defer the final decision to @palkan

File: node/node.go
Comment (palkan) [inline node/node.go:406]:
```suggestion
	n.Metrics.RegisterCounter(metricsSentMsg, "The total number of messages sent to clients")
```

File: node/node.go
Comment (palkan) [inline node/node.go:407]:
```suggestion
	n.Metrics.RegisterCounter(metricsFailedSent, "The total number of messages failed to send to clients")
```

File: node/session.go
Comment (palkan) [inline node/session.go:144]: I don't think that's the right place to increment this counter: sending to the `s.send` channel is not equal to sending to the client; we should better add this after we call `Write` (in `SendMessages`) and when it doesn't return an error.

And `metricsFailedSent` should be handled in `SendMessages` as well.

File: docs/instrumentation.md
Comment (palkan) [inline docs/instrumentation.md:76]:
```suggestion
# HELP server_msg_total The total number of messages sent to clients
# TYPE server_msg_total counter
server_msg_total 453

# HELP failed_server_msg_total The total number of messages failed to send to clients
# TYPE failed_server_msg_total counter
failed_server_msg_total 0
```

File: node/session.go
Comment (palkan) [reply node/session.go:143]: Yeah, that reveals our testing problems 😀

I've already fixed this in my _super-secret-branch_, so, let's leave this check for now.

**Review by palkan — CHANGES_REQUESTED** [review]
Thanks for the suggestion!

Could you, please, also provide a bit more context on how do you want to use this data?

File: node/session.go
Comment (prburgu) [reply node/session.go]: I've moved the code to SendMessages and also removed the nil check.

File: docs/instrumentation.md
Comment (palkan) [inline docs/instrumentation.md:70]: Oh, it should have `anycable_go_` prefix.

File: node/node.go
Comment (palkan) [inline node/node.go:407]:
```suggestion
	n.Metrics.RegisterCounter(metricsFailedSent, "The total number of messages failed to send to clients")
```

File: docs/instrumentation.md
Comment (prburgu) [reply docs/instrumentation.md]: Good catch. I've fixed this now.

**Comment by prburgu** [comment]
@palkan I've made the requested changes. This is ready for another review.

File: docs/instrumentation.md
Comment (prburgu) [reply docs/instrumentation.md]: @palkan Can I get approval on this? I've updated the description with my usecase.

**Comment by prburgu** [comment]
@palkan gentle ping

**Comment by palkan** [comment]
Merged via https://github.com/anycable/anycable-go/commit/6bc4a6f32117230b1035d52dc4d784b8ebe44b05

**Pull request closed without merging**
